src/ripple/app/tx/impl/Change.cpp

```diff
@@ -233,41 +233,67 @@ Change::applyAmendment()
     if (gotMajority && lostMajority)
         return temINVALID_FLAG;
 
-    STArray newMajorities(sfMajorities);
+    STArray majorities = amendmentObject->getFieldArray(sfMajorities);
 
-    bool found = false;
-    if (amendmentObject->isFieldPresent(sfMajorities))
+    auto it = std::find_if(
+        majorities.begin(), majorities.end(), [&amendment](STObject const& o) {
+            return o[sfAmendment] == amendment;
+        });
+
+    if (it == majorities.end() && lostMajority)
+        return tefALREADY;
+
+    if (it != majorities.end())
     {
-        const STArray& oldMajorities =
-            amendmentObject->getFieldArray(sfMajorities);
-        for (auto const& majority : oldMajorities)
+        if (gotMajority)
+            return tefALREADY;
+
+        majorities.erase(it);
+    }
```

Comment:
The first time an amendment loses majority, the program will reach line 251, remove the amendment from the `sfMajorities` array, and continue, presumably to add it to the `sfFlappingAmendments` array. The second time an amendment loses majority, the program will reach line 244 and return, never continuing to fix the `sfFlappingAmendments` array. Line 251 needs to be conditional on whether the amendment is in the `sfFlappingAmendments` array (which itself is conditional on the amendment).

There does not seem to be a new test for this new behavior. Have you added one?